src/csv_arrow_ingest.cpp

```diff
@@ -0,0 +1,98 @@
+#include <arrow/buffer.h>
+#include <arrow/io/api.h>
+#include <arrow/util/compression.h>
+#include <csv_arrow_ingest.hpp>
+#include <decryption.hpp>
+
+arrow::csv::ConvertOptions
+get_arrow_convert_options(std::vector<std::string> &utf8_columns) {
+  auto convert_options = arrow::csv::ConvertOptions::Defaults();
+  if (!utf8_columns.empty()) {
+    // read all update-file CSV columns as text to accommodate
+    // unmodified_string values
+    for (auto &col_name : utf8_columns) {
+      convert_options.column_types.insert({col_name, arrow::utf8()});
+    }
+  }
+  return convert_options;
+}
+
+template <typename T>
+std::shared_ptr<arrow::Table>
+read_csv_stream_to_arrow_table(T &input_stream,
+                               std::vector<std::string> &utf8_columns,
```

Comment:
nit: I find it convenient to have API arguments organized with "input" first (and const) and "output" last, so here (if I understand correctly), could we have:
```
read_csv_stream_to_arrow_table(const T &input_stream,
                               const std::string &filename,
                               std::vector<std::string> &utf8_columns);
```
?
(fairly minor and could be adjusted later - also applicable to other functions in this file)